src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNode.cs

```diff
@@ -39,7 +42,30 @@ public RazorDiagnosticCollection Diagnostics
         public bool HasDiagnostics => _diagnostics != null && _diagnostics.Count > 0;
 
         public SourceSpan? Source { get; set; }
-        
+
         public abstract void Accept(IntermediateNodeVisitor visitor);
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Collapsed)]
+        private string Tree
```

Comment:
This is a debugger-only property - it has no purpose in the actual code. Now you can inspect any IR node to see a dump of the tree by expanding this property.